utils/mvn/utils.go

```diff
@@ -47,13 +48,42 @@ func RunMvn(configPath, deployableArtifactsFile string, buildConf *utils.BuildCo
 	return mvnRunConfig.runCmd()
 }
 
-func validateMavenInstallation() error {
+func getMavenHome() (string, error) {
 	log.Debug("Checking prerequisites.")
 	mavenHome := os.Getenv(MavenHome)
 	if mavenHome == "" {
-		return errorutils.CheckError(errors.New(MavenHome + " environment variable is not set"))
+		// The M2_HOME environment variable is not defined.
+		// Since Maven installation can be located in different locations,
+		// Depending on the installation type and the OS (for example: For Mac with brew install: /usr/local/Cellar/maven/{version}/libexec or Ubuntu with debian: /usr/share/maven),
+		// We need to grab the location using the mvn --version command
+
+		// First we will try lo look for 'mvn' in PATH.
+		mvnPath, err := exec.LookPath("mvn")
+		if err != nil || mvnPath == "" {
+			return "", errorutils.CheckError(errors.New("No maven in PATH. "))
+		}
+		log.Debug(MavenHome, " is not defined. Retrieving Maven home using 'mvn --version' command.")
+		cmd := exec.Command("mvn", "--version")
+		var stdout bytes.Buffer
+		cmd.Stdout = &stdout
+		err = errorutils.CheckError(cmd.Run())
+		if err != nil {
+			return "", err
+		}
+		output := strings.Split(strings.TrimSpace(stdout.String()), "\n")
+		// Finding the relevant "Maven home" line in command response.
+		for _, line := range output {
+			if strings.HasPrefix(line, "Maven home:") {
+				mavenHome = strings.Split(line, " ")[2]
+				break
+			}
+		}
+		if mavenHome == "" {
+			return "", errorutils.CheckError(errors.New("Can't find maven home directory. Set 'M2_HOME' environment variable or install maven on your machine.\n\"mvn --version\" command result:\n" + stdout.String()))
```

Comment:
Can't find maven home directory. Set 'M2_HOME' environment variable or install maven on your machine.
"mvn --version" command result:
-->
Could not find the location of the maven home directory, by running 'mvn --version' command. The command output is XXX. You also have the option of setting the M2_HOME environment variable value to the maven installation directory, which is the directory which includes the bin and lib directories.